chat-client/src/client/chat.test.ts

```diff
@@ -0,0 +1,69 @@
+import { injectJSDOM } from '../test/jsDomInjector'
+// This needs to be run before all other imports so that mynah ui gets loaded inside of jsdom
+injectJSDOM()
+
+import { afterEach } from 'mocha'
+import { assert } from 'sinon'
+import { NEW_TAB_CREATED, TAB_CHANGED, TAB_REMOVED, UI_IS_READY } from '../contracts/serverContracts'
+import { SEND_TO_PROMPT, TAB_ID_RECEIVED } from '../contracts/uiContracts'
+import { createChat } from './chat'
+import sinon = require('sinon')
+
+describe('Chat', () => {
+    const sandbox = sinon.createSandbox()
+    let clientApi: { postMessage: any }
+
+    beforeEach(() => {
+        clientApi = {
+            postMessage: sandbox.stub(),
+        }
+    })
+
+    afterEach(() => {
+        sandbox.restore()
+    })
+
+    it('chat initialized publishes ready event', () => {
+        createChat(clientApi)
+        assert.calledOnceWithExactly(clientApi.postMessage, { command: UI_IS_READY })
+    })
+
+    it('send to prompt publishes telemetry event', () => {
+        createChat(clientApi)
+
+        const sendToPromptEvent = createInboundEvent({ command: SEND_TO_PROMPT, params: { prompt: 'hey' } })
+        window.dispatchEvent(sendToPromptEvent)
```

Comment:
Do integrators of the `chat-client` need to dispatch messages to `window` to communicate with it? Which `window` is that in the context of `postMessage`?

Reply:
For cross-origin, `postMessage` will work as usual.